tests/unit/plugins/inventory/test_linode.py

```diff
@@ -37,11 +37,25 @@ def test_missing_access_token_lookup(inventory):
         assert 'Could not retrieve Linode access token' in error_message
 
 
-def test_verify_file(tmp_path, inventory):
+def test_verify_file_yml(tmp_path, inventory):
```

Comment:
Can someone explain what `tmp_path` is in this context? I don't quite understand how the testing is setup.

How does `file = tmp_path / "foobar.linode.yml` get validated as a correct path but omitting the `tmp_path` does not?

Reply:
tmp_path is a pytest fixture
https://docs.pytest.org/en/latest/how-to/tmp_path.html

The tests verify the following:
- touch a validly named and validly empty files into existence. Verify returns true
- files that don't exist are not valid.
- The wrongcloud test asserts that we get False for a foo.yaml without the .linode suffix.

Reply:
tmp_path is a pytest fixture
https://docs.pytest.org/en/latest/how-to/tmp_path.html

The tests that assert the file exists touch the empty, yet validly named and validly empty, files into place. The other tests are asserting that files that don't exist are not valid.

There is not a test asserting that we get False for a foo.yaml without the .linode suffix.

Reply:
Thanks for the link to the documentation. Makes sense